### PR TITLE
FIX: Storing ``--bids-filters`` within config file

### LINKED ISSUE #2176: bids_filter null values broken by fmriprep.config
### What version of fMRIPrep are you using?

20.1.0

### What kind of installation are you using? Containers (Singularity, Docker), or "bare-metal"?

Singularity pulled from dockerhub.

### What is the exact command-line you used?
```
<place your command line here>
```
NA

### Have you checked that your inputs are BIDS valid?

NA

### Did fMRIPrep generate the visual report for this particular subject? If yes, could you share it?

NA

### Can you find some traces of the error reported in the visual report (at the bottom) or in *crashfiles*?
NA 

### Are you reusing previously computed results (e.g., FreeSurfer, Anatomical derivatives, work directory of previous run)?
NA

### fMRIPrep log
If you have access to the output logged by fMRIPrep, please make sure to attach it as a text file to this issue.

NA

## BUG

I am using a bids-filter file containing:

```
{
  "t1w": {
    "reconstruction": null,
    "acquisition": null
  }
}
```

But after runnning fMRIPrep, in the config file the corresponding section is empty:

```
[execution.bids_filters.T1w]
```

I tested `toml.dumps` with a similar dict, and it seems that it does not dump any dict entry with value None.

 ```
In [3]: toml.dumps(dict(t1w=dict(acquisition=None)))                                       
Out[3]: '[t1w]\n'                                                                        
                                                                                           
In [4]: toml.dumps(dict(t1w=dict(acquisition=1)))                                                  
Out[4]: '[t1w]\nacquisition = 1\n'                                                                   
                                                                                         
```

I don't know if this is a toml issue, or should we write an encoder/decoder that allows None/null values.

The other option would be to replace None with pybids.value (as it is done for `bids.layout.Query.ANY`) which dumps with toml:

```
In [8]: toml.dumps(dict(t1w=dict(acquisition=bids.layout.Query.NONE)))
Out[8]: '[t1w]\nacquisition = "<Query.NONE: 1>"\n'
```

There might be other cases when storing null value is significant.

### PR DESCRIPTION
**Pull request by bpinsard**
## Changes proposed in this pull request

fixes #2176

replace meaningful  bids-filter None values with bids.layout.Query.NONE to be kept with config toml dump/load.

### CODE REVIEW

**Comment by bpinsard** [comment]
Sorry, it doesn't work, the value is loaded a string.

```
In [16]: toml.loads(toml.dumps(dict(t1w=dict(acquisition=bids.layout.Query.NONE))))
Out[16]: {'t1w': {'acquisition': '<Query.NONE: 1>'}}
```

File: fmriprep/cli/parser.py [hunk 1]
```diff
@@ -48,7 +48,8 @@ def _drop_sub(value):
 
     def _filter_pybids_none_any(dct):
         import bids
-        return {k: bids.layout.Query.ANY if v == "*" else v
+        return {k: bids.layout.Query.NONE if v is None else
+                (bids.layout.Query.ANY if v == "*" else v)
                 for k, v in dct.items()}
```
Comment (oesteban) [inline fmriprep/cli/parser.py:53]: some stylistic minor:
```suggestion
        return {
            k: bids.layout.Query.NONE
            if v is None
            else (bids.layout.Query.ANY if v == "*" else v)
            for k, v in dct.items()
        }
```

File: fmriprep/config.py [hunk 2]
```diff
@@ -410,6 +410,13 @@ def init(cls):
                 ignore=("code", "stimuli", "sourcedata", "models",
                         "derivatives", re.compile(r'^\.')))
         cls.layout = cls._layout
+        if cls.bids_filters:
+            from bids.layout import Query
+            # unserialize pybids Query enum values
+            for acq, filters in cls.bids_filters.items():
+                cls.bids_filters[acq] = {
+                    k: getattr(Query, v[7:-4]) if not isinstance(v, Query) and 'Query' in v else v
+                    for k, v in filters.items()}
```
Comment (oesteban) [inline fmriprep/config.py:419]:
```suggestion
                    k: getattr(Query, v[7:-4])
                    if not isinstance(v, Query) and "Query" in v
                    else v
                    for k, v in filters.items()
                }
```

Comment (oesteban) [reply fmriprep/config.py:419]: I'm not sure I understand that `getattr(Query, v[7:-4])`.

File: fmriprep/cli/parser.py (hunk 1 above)
Comment (mgxd) [reply fmriprep/cli/parser.py:53]: +1 on this, greatly improves the readability

**Comment by mgxd** [comment]
hey @bpinsard, I applied @oesteban in b562327 - does this still work for you?

**Comment by bpinsard** [comment]
Perfect, works for me. Thanks!

**Pull request merged**